feat(registry): add remove latest partition job

bnchrch opened the pull request:
## What
Provides a job to delete all latest partitions.

Why?

If you remove the partitions they will be readded **triggering a reprocess of latest**
## Review guide
<!--
1. `x.py`
2. `y.py`
-->

## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [ ] YES 💚
- [ ] NO ❌

bnchrch commented:
* **#44795** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/44795?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

erohmensing commented on airbyte-ci/connectors/metadata_service/orchestrator/pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "orchestrator"
-version = "0.4.0"
+version = "0.4.1"
```
```suggestion
version = "0.5.0"
```
Since we added a new job

erohmensing commented on airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py:
```diff
@@ -52,6 +52,35 @@ def remove_stale_metadata_partitions():
+@op(required_resource_keys={"latest_metadata_file_blobs"})
+def remove_latest_metadata_partitions_op(context):
+    """
+    This op is responsible for removing for latest metadata files. (Generally used to reprocess metadata files).
```
```suggestion
    This op is responsible for removing for latest metadata partitions. (Generally used to reprocess metadata files).
```
The difference between these 2 is what made it so hard for me to comprehend for so long, so keeping it correct :D

erohmensing reviewed (approved):
I watched ben test this new job (and subsequent materialization) with mine own eyes. Let's do it

<img src="https://media1.giphy.com/media/4HZ473mZqE91CqlYN5/giphy.gif"/>

alafanechere reviewed (commented):
Two questions:
### Why do yo want to reprocess latest?
My understanding of the system is that partitions are per etag, so if latest changed new metadata partition will get added. If latest did not change I don't get why we'd want to recreate the partition and retrrigger all downstream operations.

### What is scheduling the job
I'm not seeing any code change related to scheduling this `remove_latest_metadata_partitions`. There's probably an implicit logic I'm unaware of.

bnchrch commented:
@alafanechere Great questions.

@erohmensing and I added this so in the event we updated the registry entry generation logic, we could manually retrigger registry entry processing.

How we do that is simply by deleting the partition keys for all latest metadata files

as it will be picked up on the next "add partition run" and processed again.

erohmensing commented:
To add - for example if we add a new (generated) field in the metadata, we want to be able to populate that info for the connectors without having to re-release them

alafanechere commented:
Sounds definitely legit. @erohmensing @bnchrch will you manually run the job to delete the partitions?

erohmensing commented:
@alafanechere Yep, the plan is to do that in the evening in SF so as not to create a backlog that slows connectors being published during the day!

bnchrch commented:
@alafanechere  As the new DAG master Ill merge when we have your approval :)

alafanechere commented:
> @alafanechere  As the new DAG master Ill merge when we have your approval :)

I'm a GitHub mobile so I don't think I can 👍 from there. But I do approve! Feel free to merge
